### ISSUE TITLE
Build for 0.18.2 fails on MacOSX

### ISSUE THREAD

**Issue by andrija-hers** [libxmljs/libxmljs#449]
```
  CC(target) Release/obj.target/libxml/vendor/libxml/xpointer.o
  LIBTOOL-STATIC Release/xml.a
  CXX(target) Release/obj.target/xmljs/src/libxmljs.o
In file included from ../src/libxmljs.cc:7:
In file included from ../src/libxmljs.h:7:
In file included from ../node_modules/nan/nan.h:196:
../node_modules/nan/nan_maybe_43_inl.h:220:17: error: 'CloneElementAt' is deprecated [-Werror,-Wdeprecated-declarations]
  return array->CloneElementAt(GetCurrentContext(), index);
                ^
/Users/andra/.node-gyp/6.5.0/include/node/v8.h:3032:36: note: 'CloneElementAt' has been explicitly marked deprecated here
                MaybeLocal<Object> CloneElementAt(Local<Context> context,
                                   ^
In file included from ../src/libxmljs.cc:7:
In file included from ../src/libxmljs.h:7:
In file included from ../node_modules/nan/nan.h:202:
In file included from ../node_modules/nan/nan_new.h:189:
../node_modules/nan/nan_implementation_12_inl.h:40:29: error: 'New' is deprecated [-Werror,-Wdeprecated-declarations]
  return v8::BooleanObject::New(value).As<v8::BooleanObject>();
                            ^
/Users/andra/.node-gyp/6.5.0/include/node/v8.h:4007:56: note: 'New' has been explicitly marked deprecated here
  V8_DEPRECATED("Pass an isolate", static Local<Value> New(bool value));
                                                       ^
In file included from ../src/libxmljs.cc:7:
In file included from ../src/libxmljs.h:7:
../node_modules/nan/nan.h:2017:15: error: 'SetAccessor' is deprecated [-Werror,-Wdeprecated-declarations]
  return obj->SetAccessor(
              ^
/Users/andra/.node-gyp/6.5.0/include/node/v8.h:2751:22: note: 'SetAccessor' has been explicitly marked deprecated here
                bool SetAccessor(Local<Name> name,
                     ^
../src/libxmljs.cc:60:17: error: 'IsDead' is deprecated [-Werror,-Wdeprecated-declarations]
    if (v8::V8::IsDead())
                ^
/Users/andra/.node-gyp/6.5.0/include/node/v8.h:8666:10: note: 'IsDead' has been explicitly marked deprecated here
bool V8::IsDead() {
         ^
4 errors generated.
```

**Comment by semyonchuryakov** [comment]
the same issue

**Comment by rchipka** [comment]
There are [several V8 deprecations](https://github.com/nodejs/nan/pull/568) included with Node v6 and later. We need to bump our NAN version.

To avoid this in the future we should use a tilde (`~2.3.2`) before the version number, which allow newer patch versions.

**Comment by rchipka** [comment]
Note: PR #450 only fixes part of this issue. 

We still need to patch libxmljs.cc so that it doesn't use the deprecated `v8::IsDead` method.

I'm not sure what the new method for this is, but once I find out I'll patch it and make a PR.

After that we can bump to libxmljs v0.18.3.

**Comment by acanimal** [comment]
@rchipka What can we do while this issue is solve? Use Node 4?

**Comment by rchipka** [comment]
@acanimal You can downgrade node-gyp to a version prior to 3.5.0, which introduced [deprecation warnings](https://github.com/nodejs/node-gyp/pull/920).

`npm install -g node-gyp@3.3.1`
`node-gyp rebuild`

**Comment by semyonchuryakov** [comment]
I solved this issue using `npm shrinkwrap`, because libxmljs is used by another node package with ^0.18.0 version. Please find below my npm-shrinkwrap.json:
`{
	"dependencies": {
		"gulp-cordova-bump": {
			"version": "1.3.1",
			"from": "gulp-cordova-bump@1.3.1",
			"dependencies": {
				"gulp-xml-transformer": {
					"version": "^1.2.0",
					"from": "^1.2.0",
					"dependencies": {
						"libxmljs": {
							"version": "0.18.0",
							"from": "^0.18.0"
						}
					}
				}
			}
		}
	}
}
`
I just added it in root project and run `npm install`. npm version is 3.10.10 and node version is 6.9.4.

**Comment by acanimal** [comment]
Maybe it can be helpful for someone.

- I'm trying to run this project [site-mapper](https://github.com/seifertd/site-mapper) which depends on `libxml@^0.18.0`, which in turn, installs 0.18.2.
- I'm working on MaxOS-Sierr
- I have downgraded `node-gyp@3.3.1  /usr/local/lib/node_modules/node-gyp` and `node@6.4.0`.

When running `npm install` I get the next error related with deprecated methods:

```
  CC(target) Release/obj.target/libxml/vendor/libxml/xmlunicode.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xmlwriter.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xpath.o
In file included from ../vendor/libxml/xpath.c:476:
../vendor/libxml/trionan.c:194:1: warning: unused function 'trio_is_special_quantity' [-Wunused-function]
trio_is_special_quantity
^
../vendor/libxml/trionan.c:218:1: warning: unused function 'trio_is_negative' [-Wunused-function]
trio_is_negative
^
2 warnings generated.
  CC(target) Release/obj.target/libxml/vendor/libxml/xpointer.o
  LIBTOOL-STATIC Release/xml.a
  CXX(target) Release/obj.target/xmljs/src/libxmljs.o
In file included from ../src/libxmljs.cc:7:
In file included from ../src/libxmljs.h:7:
In file included from ../../nan/nan.h:196:
../../nan/nan_maybe_43_inl.h:220:17: error: 'CloneElementAt' is deprecated [-Werror,-Wdeprecated-declarations]
  return array->CloneElementAt(GetCurrentContext(), index);
                ^
/Users/user/.node-gyp/6.4.0/include/node/v8.h:3012:36: note: 'CloneElementAt' has been explicitly marked deprecated here
                MaybeLocal<Object> CloneElementAt(Local<Context> context,
                                   ^
In file included from ../src/libxmljs.cc:7:
In file included from ../src/libxmljs.h:7:
In file included from ../../nan/nan.h:202:
In file included from ../../nan/nan_new.h:189:
../../nan/nan_implementation_12_inl.h:40:29: error: 'New' is deprecated [-Werror,-Wdeprecated-declarations]
  return v8::BooleanObject::New(value).As<v8::BooleanObject>();
                            ^
/Users/user/.node-gyp/6.4.0/include/node/v8.h:3976:56: note: 'New' has been explicitly marked deprecated here
  V8_DEPRECATED("Pass an isolate", static Local<Value> New(bool value));
                                                       ^
In file included from ../src/libxmljs.cc:7:
In file included from ../src/libxmljs.h:7:
../../nan/nan.h:2017:15: error: 'SetAccessor' is deprecated [-Werror,-Wdeprecated-declarations]
  return obj->SetAccessor(
              ^
/Users/user/.node-gyp/6.4.0/include/node/v8.h:2736:22: note: 'SetAccessor' has been explicitly marked deprecated here
                bool SetAccessor(Local<Name> name,
                     ^
../src/libxmljs.cc:60:17: error: 'IsDead' is deprecated [-Werror,-Wdeprecated-declarations]
    if (v8::V8::IsDead())
                ^
/Users/user/.node-gyp/6.4.0/include/node/v8.h:8462:10: note: 'IsDead' has been explicitly marked deprecated here
bool V8::IsDead() {
         ^
4 errors generated.
make: *** [Release/obj.target/xmljs/src/libxmljs.o] Error 1
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
```

To fix it I needed to fix the libxml version to 0.18.0.

**Comment by rchipka** [comment]
A fix is on the way. I'm just waiting for AppVeyor to run the tests on Windows, then I'll PR the changes and we can release a patch version.

@acanimal After a patch version is released, I suggest you revert site-mapper back to using `^` (or at least `~`) in the package.json version string.
